roc/endpoint_test.go

```diff
@@ -102,3 +103,58 @@ func TestEndpoint(t *testing.T) {
 		})
 	}
 }
+
+func TestEndpointErrors(t *testing.T) {
+	t.Run("parse uri with zero byte", func(t *testing.T) {
+		uri := "rtsp://192.168.0.1:12345\x00"
+		parseErr := errors.New("invalid URI: ")
+		endp, err := ParseEndpoint(uri)
+		require.Error(t, err)
+		require.Contains(t, err.Error(), parseErr.Error())
+		require.Nil(t, endp)
+	})
+}
+
+func Test_UriError(t *testing.T) {
```

Comment:
I think instead of adding to standalone tests TestEndpointErrors and Test_UriError we can just add cases to the table in TestEndpoint.

I suppose we'll need to modify TestEndpoint to use require.Contains for error checking instead of require.Equal.